src/coap_blockwise.c

```diff
@@ -81,18 +84,25 @@ static enum golioth_status upload_single_block(struct golioth_client *client,
                                                struct blockwise_transfer *ctx)
 {
     assert(ctx->block_size > 0 && ctx->block_size <= CONFIG_GOLIOTH_BLOCKWISE_UPLOAD_BLOCK_SIZE);
-    return golioth_coap_client_set_block(client,
-                                         ctx->path_prefix,
-                                         ctx->path,
-                                         ctx->is_last,
-                                         GOLIOTH_CONTENT_TYPE_JSON,
-                                         ctx->offset,
-                                         ctx->block_buffer,
-                                         ctx->block_size,
-                                         on_block_sent,
-                                         ctx,
-                                         true,
-                                         GOLIOTH_SYS_WAIT_FOREVER);
+    enum golioth_status err = golioth_coap_client_set_block(client,
+                                                            ctx->path_prefix,
+                                                            ctx->path,
+                                                            ctx->is_last,
+                                                            GOLIOTH_CONTENT_TYPE_JSON,
+                                                            ctx->offset,
+                                                            ctx->block_buffer,
+                                                            ctx->block_size,
+                                                            on_block_sent,
+                                                            ctx,
+                                                            false,
+                                                            GOLIOTH_SYS_WAIT_FOREVER);
+    if (GOLIOTH_OK == err)
```

Comment:
It's a defensive pattern. It you make a typo and write `=` instead of `==`, placing the constant first will cause a compiler error, while placing the variable first will compile correctly but probably isn't what you intended.